common/src/main/java/com/wynntils/core/chat/tabs/ChatTab.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.chat.tabs;
+
+import com.wynntils.core.chat.RecipientType;
+import com.wynntils.mc.event.LocalMessageEvent;
+import com.wynntils.wynn.event.ChatMessageReceivedEvent;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+public class ChatTab {
+    private String name;
+    private boolean lowPriority;
+
+    // Filters
+    private Set<RecipientType> filteredTypes;
+    private String customRegexString;
+
+    private transient Pattern customRegex;
+
+    public ChatTab(String name, boolean lowPriority, Set<RecipientType> filteredTypes, String customRegexString) {
+        this.name = name;
+        this.lowPriority = lowPriority;
+        this.filteredTypes = filteredTypes;
+        this.customRegexString = customRegexString;
+    }
+
+    public boolean matchMessageFromEvent(ChatMessageReceivedEvent event) {
+        if (filteredTypes != null && !filteredTypes.isEmpty() && !filteredTypes.contains(event.getRecipientType())) {
```

Comment:
Why would one create a chat tab like that? and more importantly, should we allow users to fall into that rabbithole?

Reply:
I see your point. Fine, keep it this way. But then maybe you should set the filteredTypes to the empty set in the constructur... except of course you can't do that because of json.... *sigh*